src/main/java/com/icthh/xm/tmf/ms/activation/service/SagaServiceImpl.java

```diff
@@ -145,7 +147,10 @@ private void doTask(SagaEvent sagaEvent, Context context) {
             StopWatch stopWatch = StopWatch.createStarted();
             log.info("Start execute task by event {} transaction {}", sagaEvent, transaction);
             Continuation continuation = new Continuation();
+            Set<String> nextTasks =  new HashSet<>(taskSpec.getNext());
             Map<String, Object> taskContext = taskExecutor.executeTask(taskSpec, sagaEvent, transaction, continuation);
+            nextTasks.removeAll(taskSpec.getNext());
+            markAsRejectedByCondition(nextTasks, transaction);
```

Comment:
add test that check transaction finished after branch

Reply:
Done